### PR TITLE
feat(KafkaTracker): Adding kafka tracker to red cross

### PR DESCRIPTION
**Pull request by RonBarabash**
https://jira.yotpo.com/browse/YO-3312
with 
https://github.com/YotpoLtd/yotpo-api/pull/1768

### CODE REVIEW

File: lib/red_cross.rb [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
-        Configuration.segment.flush
+        Configuration.tracker.flush
       end
 
       def monitor(attrs)
         []
       end
+
+      def method_missing(m, *args, &block)
+        arr = m.to_s.split("_")
```
Comment (RonBarabash) [inline lib/red_cross.rb]: Need to add some sort of validation?

File: lib/red_cross/configuration.rb [hunk 2]
```diff
@@ -1,19 +1,12 @@
 module RedCross
   module Configuration
     class << self
-      attr_accessor :path, :segment_write_key, :segment, :influxdb_connect
+      attr_accessor :influxdb_connect, :influxdb, :trackers, :default, :tracker
```
Comment (RonBarabash) [inline lib/red_cross/configuration.rb]: ?
Default Tracker

File: lib/red_cross/trackers/kafka_tracker.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            body: method == :post ? params : {}
+        )
+        request.on_complete do |response|
+          if response.success?
+            return response.body
+          else
+            raise "Failed pushing to Kafka. Error code: #{response.response_code}, route: #{request_url}"
+          end
+        end
+        request.run
```
Comment (RonBarabash) [inline lib/red_cross/trackers/kafka_tracker.rb]: Implement empty flush and identify

File: lib/red_cross/trackers/kafka_tracker.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      def clerk_request(request_url, params, method = :get)
+        request = Typhoeus::Request.new(
+            request_url,
+            method: method,
+            params: method == :get ? params : {},
+            body: method == :post ? params : {}
+        )
+        request.on_complete do |response|
+          if response.success?
+            return response.body
```
Comment (RonBarabash) [inline lib/red_cross/trackers/kafka_tracker.rb]: Check the return types for all trackers are the same

File: lib/red_cross/trackers/kafka_tracker.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          else
+            raise "Failed pushing to Kafka. Error code: #{response.response_code}, route: #{request_url}"
```
Comment (RonBarabash) [inline lib/red_cross/trackers/kafka_tracker.rb]: Create new exception and raise it here

File: lib/red_cross/trackers/kafka_tracker.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+module RedCross
+  module Trackers
+    class KafkaTracker < RedCross::Trackers::Base
+      attr_accessor :clerk_host
+
+      def initialize(host)
+        @clerk_host = host
+      end
+
+      def track(attrs, topic = "")
```
Comment (RonBarabash) [inline lib/red_cross/trackers/kafka_tracker.rb]: change topic to a default empty hash

**Comment by RonBarabash** [comment]
@avichay finish with code review changes

File: lib/red_cross.rb [hunk 7]
```diff
@@ -1,26 +1,36 @@
...
-        Configuration.segment.flush
+        Configuration.tracker.flush
       end
 
       def monitor(attrs)
         []
       end
+
+      def method_missing(m, *args, &block)
+        match = /(.*?)_track/.match(m.to_s)
```
Comment (RonBarabash) [inline lib/red_cross.rb:30]: @avichay we can do it like this but this way the method is not dynamic, were sending only :track

Comment (avichay) [inline lib/red_cross.rb:30]: @ronbarab but currently this is the only thing we support no ?

Comment (RonBarabash) [inline lib/red_cross.rb:30]: @avichay yep

**Comment by RonBarabash** [comment]
@avichay if theres nothing else please pass to QA

**Comment by avichay** [comment]
!ninjastar

**Comment by dmitri86git** [comment]
@NetanelMoshkovitz Approved

**Pull request merged**